x-pack/plugins/apm/server/routes/transactions.ts

```diff
@@ -127,19 +127,27 @@ export function initTransactionsApi(server: Server) {
       pre,
       validate: {
         query: withDefaultValidators({
-          transaction_name: Joi.string().required()
+          transaction_name: Joi.string().required(),
+          transaction_id: Joi.string().default('')
         })
       }
     },
     handler: req => {
       const { setup } = req.pre;
       const { serviceName } = req.params;
-      const { transaction_name: transactionName } = req.query as {
+      const {
+        transaction_name: transactionName,
+        transaction_id: transactionId
+      } = req.query as {
         transaction_name: string;
+        transaction_id: string;
       };
```

Comment:
The typescript destructuring can be a bit harder to read. Could be made simple to replace the above 7 lines with the following 2 lines:
```
const transactionName: string = query.transaction_name;
const transactionId: string = query.transaction_id;
```